extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -237,18 +237,14 @@ function createDicomWebApi(dicomWebConfig, servicesManager) {
           };
 
           const denaturalized = denaturalizeDataset(meta);
-          let dicomDict = new DicomDict(denaturalized);
 
-          dicomDict.dict = denaturalizeDataset(dataset);
+          const defaultDicomDict = new DicomDict(denaturalized);
 
-          const onBeforeDicomStore =
-            customizationService.getModeCustomization('onBeforeDicomStore')?.value;
+          defaultDicomDict.dict = denaturalizeDataset(dataset);
 
-          if (typeof onBeforeDicomStore === 'function') {
-            dicomDict = onBeforeDicomStore({ dicomDict, measurementData, dataset });
-          }
+          const effectiveDicomDict = dicomDict ?? defaultDicomDict;
```

Comment:
You can completely skip creating a newDicomDict if the dicomDict is passed in correctly.

Reply:
@sedghi please check it again.